### ISSUE TITLE
Notification only work on DebugJS mode ?

### ISSUE THREAD

**Issue by hai2014bk** [invertase/react-native-firebase#1015]
### Environment

iOS 

3. Build Tools:

4. React Native version:

0.52

5. RNFirebase Version:

4.0.5

Hi guys, thanks for amazing source, but i'm facing this: 

```
const notificationOpen: NotificationOpen = await firebase.notifications().getInitialNotification();
        if (notificationOpen) {
            const action = notificationOpen.action;
            const notification: Notification = notificationOpen.notification;
            this.handleNotif(notification);
        }
        this.notificationListener = firebase.notifications().onNotification((notification: Notification) => {
            console.log('NOTIFICATION FIREBASE', notification, this);
            setTimeout(() => {
                if (this) {
                    this.props.getNotification(this.state.userData.id);
                    if (this.alert) {
                        this.alert.show({
                            title: notification._title,
                            message: notification._body,
                            onPress: () => this.handleNotif(notification)
                        });
                    }
                }
            }, 500);
        });
        this.notificationOpenedListener = firebase
            .notifications()
            .onNotificationOpened((notificationOpen: NotificationOpen) => {
                // Get the action triggered by the notification being opened
                const action = notificationOpen.action;
                // Get information about the notification that was opened
                const notification: Notification = notificationOpen.notification;
                this.handleNotif(notification);
            });
```

This is my Listener, and when i run with DebugJs, it work perfectly. But if I turn off Debug or build for Release, It not event get fired and I don't know why ??
So can you help me pls ?

**Comment by chrisbianca** [comment]
When you say this works in Debug, what does working mean?

By the looks of your code, you're using `this.alert` to indicate a notification arrives, and I'm guessing `this.alert` isn't available within release or non-debug mode.

Take a look at the build in React Native Alert: https://facebook.github.io/react-native/docs/alert.html instead.

**Comment by hai2014bk** [comment]
Sorry, that my mistake when copy it, but even i remove it (actually my code doesn't have it), all my listener are not fired. If my app is in background (or kill), i can receive notification but when i touch to banner, my listener not fired to do my logic. If app foreground, nothing happen.
I try to debug on Xcode, the `didReceiveRemoteNotification` fired, but my listener in React-Native stay down.

But, If i turn on DebugJS Remote mode on my app (by shake the app and chose), all work fine !
@chrisbianca

**Comment by chrisbianca** [comment]
Closing in favour of discord chat.

**Comment by petetastic** [comment]
Hi Chris,

Has this issue been resolved? I'm having a similar issue, where iOS notifications work fine with debugging enabled, but not when it is disabled.

Would you mind letting me know the outcome of your conversation with @hai2014bk?
Thanks,

Pete

**Comment by tal987** [comment]
@petetastic did you solve it? the function firebase.notifications().onNotificationOpened triggered only when the app is open in background (in debug mode)

**Comment by petetastic** [comment]
Yes, although I'm afraid I couldn't tell you exactly how. I ended up changing quite a few things and now it works in release mode. I can say that I learned two things: firstly I don't think it was ultimately to do with RNFirebase—I actually think that the bug I had had to do with react-navigation. But secondly and more importantly I learned that the implementation of Javascript is subtly different when using the debugger since it uses Chrome's V8 javascript engine which talks back to the app, instead of React Native's javascript (JavaScriptCore). This makes it quite hard to rely on the debugger, since it can introduce bugs that are only a function if itself, as it were. More info here: https://blog.swmansion.com/hunting-js-memory-leaks-in-react-native-apps-bd73807d0fde and here: http://www.albertgao.xyz/2018/05/30/24-tips-for-react-native-you-probably-want-to-know/

Pete
